repository_pool.go

```diff
@@ -276,7 +276,17 @@ func (i *rowRepoIter) rowReader(num int) {
 	defer i.wg.Done()
 
 	for repo := range i.repos {
-		iter, _ := i.iter.NewIterator(repo)
+		iter, err := i.iter.NewIterator(repo)
+		if err != nil {
+			// guard from possible previous error
+			select {
+			case <-i.done:
+				return
+			default:
+				i.setError(err)
+				close(i.done)
```

Comment:
You need to `continue` to the next loop iteration. Otherwise, we'll still get a panic because of the `iter == nil` and never get the error.